test/integration/ethereum/flashMintNAV.spec.ts

```diff
@@ -0,0 +1,548 @@
+  wan_liang: "0xCcb12611039c7CD321c0F23043c841F1d97287A5", // cUSDCv3
+  mane_lee: "0xBF370B6E9d97D928497C2f2d72FD74f4D9ca5825", // aUSDC
+  morpho_seeding: "0x6ABfd6139c7C3CC270ee2Ce132E309F59cAaF6a2", // gtUSDC,
+  weth: "0x4F4495243837681061C4743b74B3eEdf548D56A5",
+  dai: "0x3B5fb9d9da3546e9CE6E5AA3CCEca14C8D20041e",
+  usdt: "0xEEA81C4416d71CeF071224611359F6F99A4c4294",
+};
+
+const swapDataEmpty: SwapData = {
+  exchange: Exchange.None,
+  fees: [],
+  path: [],
+  pool: ADDRESS_ZERO,
+};
+
+const swapDataUsdcToWeth: SwapData = {
+  exchange: Exchange.UniV3,
+  fees: [500],
+  path: [addresses.tokens.USDC, addresses.tokens.weth],
+  pool: ADDRESS_ZERO,
+};
+
+const swapDataWethToUsdc = {
+  exchange: Exchange.UniV3,
+  fees: [500],
+  path: [addresses.tokens.weth, addresses.tokens.USDC],
+  pool: ADDRESS_ZERO,
+};
+
+if (process.env.INTEGRATIONTEST) {
+  describe.only("FlashMintNAV - Integration Test", async () => {
+    let owner: Account;
+    let deployer: DeployHelper;
+    let setV2Setup: SetFixture;
+    let erc4626Oracle: ERC4626Oracle;
+    let setToken: SetToken;
+    let flashMintNAV: FlashMintNAV;
+    let usdc_erc20: IERC20;
+    let aEthUSDC_erc20: IERC20;
+    let cUSDCv3_erc20: IERC20;
+    let aUSDC_erc20: IERC20;
+    let gtUSDC_erc20: IERC20;
+
+    setBlockNumber(20528609, true);
+
+    before(async () => {
+      [owner] = await getAccounts();
+
+      // Sytem setup
+      deployer = new DeployHelper(owner.wallet);
+      setV2Setup = new SetFixture(ethers.provider, owner.address);
+      await setV2Setup.initialize();
+
+      // Token setup
+      usdc_erc20 = IERC20__factory.connect(addresses.tokens.USDC, owner.wallet);
+      aEthUSDC_erc20 = IERC20__factory.connect(tokenAddresses.aEthUSDC, owner.wallet);
+      cUSDCv3_erc20 = IERC20__factory.connect(tokenAddresses.cUSDCv3, owner.wallet);
+      aUSDC_erc20 = IERC20__factory.connect(tokenAddresses.aUSDC, owner.wallet);
+      gtUSDC_erc20 = IERC20__factory.connect(tokenAddresses.gtUSDC, owner.wallet);
+
+      // Oracle setup
+      await setV2Setup.priceOracle.editMasterQuoteAsset(tokenAddresses.usdc);
+
+      const preciseUnitOracle = await deployer.setV2.deployPreciseUnitOracle("Rebasing USDC Oracle");
+      await setV2Setup.priceOracle.addAdapter(preciseUnitOracle.address);
+      await setV2Setup.priceOracle.addPair(tokenAddresses.usdc, tokenAddresses.usdc, preciseUnitOracle.address);
+      await setV2Setup.priceOracle.addPair(tokenAddresses.aEthUSDC, tokenAddresses.usdc, preciseUnitOracle.address);
+      await setV2Setup.priceOracle.addPair(tokenAddresses.cUSDCv3, tokenAddresses.usdc, preciseUnitOracle.address);
+      await setV2Setup.priceOracle.addPair(tokenAddresses.aUSDC, tokenAddresses.usdc, preciseUnitOracle.address);
+      erc4626Oracle = await deployer.setV2.deployERC4626Oracle(
+        tokenAddresses.gtUSDC,
+        usdc(1),
+        "gtUSDC - USDC Calculated Oracle",
+      );
```

Comment:
noting this constructor's arguments may change soon https://github.com/IndexCoop/index-protocol/pull/49